libcxx/include/__memory/uninitialized_algorithms.h

```diff
@@ -638,7 +638,7 @@ __uninitialized_allocator_relocate(_Alloc& __alloc, _Tp* __first, _Tp* __last, _
     __guard.__complete();
     std::__allocator_destroy(__alloc, __first, __last);
   } else {
-    __builtin_memcpy(__result, __first, sizeof(_Tp) * (__last - __first));
+    __builtin_memcpy((void*)__result, __first, sizeof(_Tp) * (__last - __first));
```

Comment:
I think it would be better to make these `static_cast`s to make it obvious that we're casting from a different pointer. Could you also add a comment like `// casting to void* to suppress clang complaining that this is technically UB`?

Reply:
done!